frontend/src/pages/login.tsx

```diff
@@ -1,19 +1,69 @@
+import { signInWithEmailAndPassword } from "firebase/auth";
+import { AlertCircle } from "lucide-react";
 import Image from "next/image";
-import { ReactElement } from "react";
+import { useRouter } from "next/navigation";
+import { ReactElement, useState } from "react";
 import { FieldValues, SubmitHandler, useForm } from "react-hook-form";
 
+import { GET } from "@/api/requests";
 import Landing from "@/components/Landing";
 import { Textfield } from "@/components/Textfield";
 import { Button } from "@/components/ui/button";
+import { initFirebase } from "@/firebase/firebase";
 import { useWindowSize } from "@/hooks/useWindowSize";
 import { cn } from "@/lib/utils";
 
 export default function Login() {
-  const { register, setValue, handleSubmit } = useForm();
+  const {
+    register,
+    setValue,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
   const _setValue = setValue;
 
+  const [firebaseError, setFirebaseError] = useState("");
+
+  const { auth } = initFirebase();
+  const router = useRouter();
+
+  const login = async (email: string, password: string) => {
+    return await signInWithEmailAndPassword(auth, email, password)
+      .then((userCredential) => {
+        return userCredential.user.getIdToken();
+      })
+      .then((token) => {
+        return token;
+      });
+  };
+
+  const sendTokenToBackend = async (token: string) => {
+    try {
+      const response = await GET(`/user/`, {
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      });
+
+      if (response.ok) {
+        console.log(await response.json());
+      } else {
+        console.error("Failed to get user info from JWT Token");
+      }
+    } catch (error) {
+      console.error("error sending JWT token to backend", error);
+    }
+  };
+
   const onSubmit: SubmitHandler<FieldValues> = (data) => {
     console.log(data);
+    login(data.email as string, data.password as string)
+      .then((token: string) => {
+        void sendTokenToBackend(token);
+      })
+      .catch((_) => {
+        setFirebaseError("Invalid login. Please check your username and password.");
+      });
+    router.push("/home");
```

Comment:
When I tried logging in with an invalid username and password it still routed to the home page.